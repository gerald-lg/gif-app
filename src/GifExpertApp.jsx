import React, { useState } from 'react'
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {


    const [categories, setCategories] = useState([ 'Haikyuu' ]);

    const onAddCategory = (input) => {

        if(categories.includes(input)) return;
        
        setCategories(cat => [ input , ...cat ]);
    }

    return (
        <>
        
             <div className="waviy title__center">
                <span style={{'--i': 1}}>G</span>
                <span style={{'--i': 2}}>I</span>
                <span style={{'--i': 3}}>F</span>
                <span style={{'--i': 4}}>E</span>
                <span style={{'--i': 5}}>X</span>
                <span style={{'--i': 6}}>P</span>
                <span style={{'--i': 7}}>E</span>
                <span style={{'--i': 8}}>R</span>
                <span style={{'--i': 9}}>T</span>
                <span style={{'--i': 10}}>A</span>
                <span style={{'--i': 11}}>P</span>
                <span style={{'--i': 12}}>P</span>
            </div>

            {/* // si lleva un "on" significa que esta emitiendo algo */}
            <AddCategory onNewCategory= { onAddCategory }/>

            { 
                categories.map(( category ) =>
                    <GifGrid key={ category } category={ category } />
                )
            }

        </>
    )
}
