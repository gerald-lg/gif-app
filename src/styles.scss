 @import '../sass/base/base';
 @import '../sass/components/grid';
 @import '../sass/components/cards';
 @import '../sass/components/spinner';
 @import '../sass/components/title';
  
input {
  background-color: white;
  border-radius: 5px;
  border: 2px solid #00ADB5;
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 100%;
}
  
h2 {
  font-size: 1.5rem;
}

h3 {
  font-size: 3rem;
  margin-bottom: 5px;
}
