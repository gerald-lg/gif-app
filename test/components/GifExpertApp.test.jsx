import { render, screen } from "@testing-library/react"
import { GifExpertApp } from "../../src/GifExpertApp"

describe('Pruebas en el componente GifExpertApp', () => { 
    
    test('Snapshot del componente', () => { 

        const { container } = render(<GifExpertApp />)
        expect(container).toMatchSnapshot();
    });

    test('Debe retornar mas de una categoria', () => { 

        render(<GifExpertApp />);

        const onAddCategory = jest.fn();
        console.log(onAddCategory);
        screen.debug();

    });

 });