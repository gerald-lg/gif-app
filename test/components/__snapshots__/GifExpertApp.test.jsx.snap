// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifExpertApp Snapshot del componente 1`] = `
<div>
  <div
    class="waviy title__center"
  >
    <span
      style="--i: 1;"
    >
      G
    </span>
    <span
      style="--i: 2;"
    >
      I
    </span>
    <span
      style="--i: 3;"
    >
      F
    </span>
    <span
      style="--i: 4;"
    >
      E
    </span>
    <span
      style="--i: 5;"
    >
      X
    </span>
    <span
      style="--i: 6;"
    >
      P
    </span>
    <span
      style="--i: 7;"
    >
      E
    </span>
    <span
      style="--i: 8;"
    >
      R
    </span>
    <span
      style="--i: 9;"
    >
      T
    </span>
    <span
      style="--i: 10;"
    >
      A
    </span>
    <span
      style="--i: 11;"
    >
      P
    </span>
    <span
      style="--i: 12;"
    >
      P
    </span>
  </div>
  <form
    aria-label="form"
    method="post"
  >
    <input
      class="custom__input"
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <div
    class="card-grid"
  >
    <span
      class="loader"
    />
  </div>
  <h1
    class="title__center"
  >
    Haikyuu
  </h1>
  <div
    class="card-grid"
  />
</div>
`;
